doc/tutorial/machine_learning_map/pyparsing.py

```diff
@@ -1354,8 +1354,8 @@ def _parseNoCache( self, instring, loc, doActions=True, callPreParse=True ):
             try:
                 try:
                     loc,tokens = self.parseImpl( instring, preloc, doActions )
-                except IndexError:
-                    raise ParseException( instring, len(instring), self.errmsg, self )
+                except IndexError as e:
+                    raise ParseException( instring, len(instring), self.errmsg, self ) from e

```

Comment:
I do not think we should modify `doc/tutorial/machine_learning_map/pyparsing.py` because it is a vendored file that is not part of the sklearn library.

Reply:
Agreed and fixed.